syntax = "proto3";

package LAMARCK_ML;

import "LAMARCK_ML/data_util/Attribute.proto";

enum ReduceProto {
    MEAN = 0;
    MINIMUM = 1;
    MAXIMUM = 2;
    SUM = 3;
    PRODUCT = 4;
    VARIANCE = 5;
    STD = 6;
}

message LossProto {
    string cls_name = 1;
    string id_name = 2;
    ReduceProto reduction = 5;
    repeated AttributeProto attr = 10;
}